### ISSUE TITLE
Missing index on transaction amount?

### ISSUE THREAD

**Issue by scholtz** [algorand/indexer#578]
## Summary
I am trying to search the blockchain by transaction amount.

https://algoexplorerapi.io/idx2/v2/transactions?currency-greater-than=300000000&currency-less-than=300000000&limit=1

To get first transaction with specific amount takes so long it triggers the timeout. << Bug

![image](https://user-images.githubusercontent.com/1223439/126062115-9e0021e7-d36c-4e35-8993-343362f61539.png)

Btw, why there is no method to search for specific amount?

I believe there is the index missing in the postgresql database.. The request should be fast. If there is the index, perhaps the postgresql is not good enough choice?

## Scope/Requirements
Indexer search for transactions

## Urgency/Relative Priority
Someone could make the DDOS requests using the ineffecient search reqests.. 

I do not consider it as urgent, but indexer providers should consider this as resource consuming..

### Acceptance criteria
There are two options here: either have index missing error reported or create an index (this should be done locally by the user)

**Comment by fabrice102** [comment]
The issue seems to be that `greater-than` and `less-than` are strict.

So you need: https://algoexplorerapi.io/idx2/v2/transactions?currency-greater-than=299999999&currency-less-than=300000001&limit=1

The answer to that query is immediate:

```bash
$ curl -s "https://algoexplorerapi.io/idx2/v2/transactions?currency-greater-than=299999999&currency-less-than=300000001&limit=1" | jq
{
  "current-round": 15091361,
  "next-token": "Cp0CAAAAAAAAAAAA",
  "transactions": [
    {
      "close-rewards": 0,
      "closing-amount": 0,
      "confirmed-round": 171274,
      "fee": 1000,
      "first-valid": 171272,
      "genesis-hash": "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=",
      "id": "S4EJKLYOQ4H3ENKVDLUK57MJVPGM6WVTBZC5M2TQRF4A3OD3XZ5Q",
      "intra-round-offset": 0,
      "last-valid": 172272,
      "payment-transaction": {
        "amount": 300000000,
        "close-amount": 0,
        "receiver": "DPHTQBLS3ZOTCLWZ24C7NHTOKX67IF7B2QFFIZOEVXXNCO7CRQXVM4ORKE"
      },
      "receiver-rewards": 0,
      "round-time": 1561078125,
      "sender": "BCQXJXINAK56QQ2H4J4RTHBJLQ6CW7ADMB6KD7CDDZ3FFRLLZGLH6VFO3U",
      "sender-rewards": 765450,
      "signature": {
        "sig": "bAAQSK4UKjRF8UJC+0VkdeRMrx8YW6kDeGVQWSbozXxiQZPpp9NshENETvUtNrNI0hnGDqBLpZapsB44LbPTCQ=="
      },
      "tx-type": "pay"
    }
  ]
}
```

However, I agree that it would be great that such an impossible query (asking for an amount that is larger and smaller than the same integer) either immediately returns no result or returns an error.

I tested it on my personal indexer and I have the same issue as with algoexplorer.

Version: https://github.com/algorand/indexer/commit/8157a43d20de8dacae34bc8d7d18b4a46346672e
i.e., indexer 2.5.0 with the PR patch https://github.com/algorand/indexer/pull/554

**Comment by scholtz** [comment]
Great, thanks.. it might work.. what about also some error when there is no transaction there?

This is valid requests but timeouts:

https://algoexplorerapi.io/idx2/v2/transactions?currency-greater-than=299999997&currency-less-than=299999999&limit=1

**Comment by winder** [comment]
This query should be included with #448, #669

**Comment by winder** [comment]
These slow queries are now disabled by default.
